doozer/doozerlib/backend/konflux_image_builder.py

```diff
@@ -0,0 +1,319 @@
+    }
+
+    def __init__(self, config: KonfluxImageBuilderConfig, logger: Optional[logging.Logger] = None) -> None:
+        self._config = config
+        self._logger = logger or LOGGER
+
+    async def build(self, metadata: ImageMetadata):
+        """ Build a container image with Konflux. """
+        # Load the build source repository
+        dest_dir = self._config.base_dir.joinpath(metadata.qualified_key)
+        build_repo = await BuildRepo.from_local_dir(dest_dir, self._logger)
+
+        if not build_repo.https_url.startswith("https://github.com/openshift-priv/"):
+            raise ValueError("Only GitHub repositories in the openshift-priv organization are supported")
+
+        # Load the Kubernetes configuration
+        cfg = Configuration()
+        config.load_kube_config(config_file=self._config.kubeconfig, context=self._config.context,
+                                persist_config=False, client_configuration=cfg)
+        with client.ApiClient(configuration=cfg) as api_client:
+            dyn_client = DynamicClient(api_client)
+
+            pipelinerun = await self._start_build(metadata, build_repo, dyn_client)
+
+            pipelinerun_name = pipelinerun['metadata']['name']
+            self._logger.info("Waiting for PipelineRun %s to complete...", pipelinerun_name)
+            pipelinerun = await self._wait_for_pipelinerun(dyn_client, pipelinerun_name)
+            if pipelinerun.status.conditions[0].status != "True":
+                raise KonfluxImageBuildError(f"Konflux image build for {metadata.distgit_key} failed", pipelinerun_name, pipelinerun)
+
+        return pipelinerun_name, pipelinerun
+
+    async def _start_build(self, metadata: ImageMetadata, build_repo: BuildRepo, dyn_client: DynamicClient):
+        git_url = build_repo.https_url
+        git_branch = build_repo.branch
+        assert build_repo.commit_hash is not None, "git_commit is required for Konflux image build"
+        git_commit = build_repo.commit_hash
+
+        df_path = build_repo.local_dir.joinpath("Dockerfile")
+        df = DockerfileParser(str(df_path))
+        version = df.labels["version"]
+        if not version:
+            raise ValueError("Dockerfile must have a 'version' label")
+        release = df.labels["release"]
+        if not release:
+            raise ValueError("Dockerfile must have a 'release' label")
+
+        # Ensure the Application resource exists
+        app_name = self._config.group_name.replace(".", "-")
+        app_manifest = self._new_application(app_name, app_name)
+        app = await self._create_or_patch(dyn_client, app_manifest)
+        self._logger.info(f"Created application: {app['metadata']['name']}")
+
+        # Ensure the component resource exists
+        component_name = f"{app_name}-{metadata.distgit_key}"
+        dest_image_repo = self._config.output_repo
+        # dest_image_tag = f"{metadata.distgit_key}_{self._uuid_tag}"
+        dest_image_tag = f"{version}-{release}"
```

Comment:
why not name-version-release?

Reply:
You are right. Initially by default config each image gets an separate repo. Since we are going to use a mono repo for all images, we need to include the image name.